### PR TITLE
[ThumbTrack] Adds the Ripple behavior to Thumbtrack.

### LINKED ISSUE #7378: [Slider] Integrate Ripple in Slider
This was filed as an internal issue. If you are a Googler, please visit [b/132671294](http://b/132671294) for more details.

---
#### Internal data
- Associated internal bug: [b/132671294](http://b/132671294)

### PR DESCRIPTION
**Pull request by yarneo**
## Related links
* Bug: Closes #7378  
* Ripple: [MDCRippleView](https://github.com/material-components/material-components-ios/tree/develop/components/Ripple)

## Introduction
This PR integrates [`MDCRippleView`](https://github.com/material-components/material-components-ios/blob/develop/components/Ripple/src/MDCRippleView.h) into [`MDCThumbTrack`](https://github.com/material-components/material-components-ios/tree/develop/components/private/ThumbTrack). This behavior is an opt in flag property added to MDCThumbTrack, `enableRippleBehavior`.  This improves the visual touch feedback for our users as MDCInkView didn't support our current motion guidelines. We are also marking any of the Ink APIs ToBeDeprecated as part of the [Deprecation Policy](https://github.com/material-components/material-components-ios/blob/cb6b8e9599e9018c545e0011a3ded82597988106/contributing/deprecation_policy.md#step-4-add-a-comment-to-the-api-indicating-that-it-will-be-deprecated).
As you can see in the below videos, Ink was never implemented correctly for Slider, and therefore never appeared. With Ripple enabled, we finally get the right results, as seen [in our spec](https://material.io/design/components/sliders.html#continuous-slider).

## Videos
| Before | After |
| - | - |
|oldRipple|newRipple|

### CODE REVIEW

File: components/Ripple/BUILD [hunk 1]
```diff
@@ -43,7 +43,7 @@ mdc_objc_library(
     name = "private",
     hdrs = native.glob(["src/private/*.h"]),
     includes = ["src/private"],
-    visibility = ["//visibility:private"],
+    visibility = ["//:__subpackages__"],
```
Comment (jverkoey) [inline components/Ripple/BUILD]: Why is this necessary?

Comment (yarneo) [reply components/Ripple/BUILD]: This is part of the ThumbTrack Ripple Testing:
Checking if MDCRippleView has an activeRippleLayer (which is an instance of class MDCRippleLayer), helps identify if invoking `touchesBegan:touches` on the ThumbTrack initializes a ripple layer to show the ripple or not. To be able to check activeRippleLayer, I need to import MDCRippleLayer which is under private.
I could just expose MDCThumbTrack in the visibility, but i assumed __subpackages__ allows any future MaterialComponent to use this BUILD target to use it for testing.

Comment (jverkoey) [reply components/Ripple/BUILD]: This visibility is a bit too open; it will allow any target in MDC make use of private implementation details. Can we restrict this to just the test target in some manner?

Comment (yarneo) [reply components/Ripple/BUILD]: I removed the dependency entirely, and used `@class MDCRippleLayer` instead. There was no need for me to get the entire class just to check nullability.

File: components/private/ThumbTrack/src/MDCThumbTrack.h [hunk 2]
```diff
@@ -143,9 +158,15 @@
 /** The max radius of the ripple when the user touches the thumb. */
 @property(nonatomic, assign) CGFloat thumbMaxRippleRadius;
 
+/** The max radius of the ripple when the user touches the thumb. */
+@property(nonatomic, assign) CGFloat thumbRippleMaximumRadius;
+
 /** Whether the thumb should display ink splashes on touch. */
 @property(nonatomic, assign) BOOL shouldDisplayInk;
```
Comment (jverkoey) [inline components/private/ThumbTrack/src/MDCThumbTrack.h]: Consider marking these apis as to be deprecated as well.

**Review by jverkoey — APPROVED** [review]

File: components/Ripple/BUILD (hunk 1 above)
Comment (jverkoey) [reply components/Ripple/BUILD]: 💯
